File: core/src/com/infeez/simple/base/GameSpriteBatch.java
package com.infeez.simple.base;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.infeez.simple.entity.GameObject;

public class GameSpriteBatch extends SpriteBatch {

    public void draw(GameObject gameObject){
        draw(gameObject.getTextureRegion(), gameObject.getX(), gameObject.getY(), gameObject.getWidth(), gameObject.getHeight());
    }

}

File: core/src/main/kotlin/com/infeez/simple/base/GameSpriteBatch.kt
package com.infeez.simple.base

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.infeez.simple.entity.GameObject

class GameSpriteBatch : SpriteBatch() {
    fun draw(gameObject: GameObject) {
        val textureRegion = gameObject.textureRegion ?: return
        draw(textureRegion, gameObject.x, gameObject.y, gameObject.width, gameObject.height)
    }
}
